src/transaction.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  greedyCoinSelector,
+  UtxoInterface,
+  networks,
+  address,
+  RecipientInterface,
+} from 'ldk';
+import { confidential, Psbt } from 'liquidjs-lib';
+
+// SwapTransactionInterface defines the minimum needed for implementation to hold data to construct a valid swap transaction.
+interface SwapTransactionInterface {
+  network: networks.Network;
+  pset: Psbt;
+  inputBlindingKeys: Record<string, Buffer>;
+  outputBlindingKeys: Record<string, Buffer>;
+  blindingPrivKeyHexByScript: (script: string) => string;
+}
+
+// SwapTransaction holds a pset and expose a create method to select coins and build a transaction for a SwapRequest message
+export class SwapTransaction implements SwapTransactionInterface {
+  network: networks.Network;
+  pset: Psbt;
+  inputBlindingKeys: Record<string, Buffer> = {};
+  outputBlindingKeys: Record<string, Buffer> = {};
+  blindingPrivKeyHexByScript: (script: string) => string;
+
+  constructor(
+    chain: string,
+    blindingPrivKeyHexByScript: (script: string) => string
+  ) {
+    this.network = (networks as any)[chain];
+    this.pset = new Psbt({ network: this.network });
+    this.blindingPrivKeyHexByScript = blindingPrivKeyHexByScript;
+  }
+
+  create(
+    unspents: Array<UtxoInterface>,
+    amountToBeSent: number,
+    amountToReceive: number,
+    assetToBeSent: string,
+    assetToReceive: string,
+    addressForSwapOutput: string,
+    addressForChangeOutput: string
+  ) {
+    // We default to dumb coinselection. ie. se sort for value, first unspent found tha covers target, we use it.
+    const coinSelector = greedyCoinSelector();
+
+    const { selectedUtxos, changeOutputs } = coinSelector(
+      unspents,
+      [
+        {
+          value: amountToBeSent,
+          asset: assetToBeSent,
+          address: '',
```

Comment:
Is this the correct way to use it? I didn't see any other way for this use-case, wher I know the target amount&asset, but not the output, since is the other swap responder that will add it

Reply:
@louisinger

Reply:
Yes, in fact the "targeted address" is not used by greedy coin selector. so it should work with and empty string